src/Account.sol

```diff
@@ -930,13 +946,26 @@ contract Account is IAccount, Auth, OpsReady {
     function _sUSDRate(IPerpsV2MarketConsolidated _market)
         internal
         view
-        returns (uint256)
+        returns (uint256, PriceOracleUsed)
     {
-        (uint256 price, bool invalid) = _market.assetPrice();
-        if (invalid) {
-            revert InvalidPrice();
+        bytes32 assetId = _market.baseAsset();
+
+        /// @dev can revert if assetId is invalid OR there's no price for the given asset
+        (uint256 price, uint256 publishTime) =
+            PERPS_V2_EXCHANGE_RATE.resolveAndGetLatestPrice(assetId);
+
+        // resolveAndGetLatestPrice is provide by pyth
+        PriceOracleUsed priceOracle = PriceOracleUsed.PYTH;
+
+        // if the price is stale, get the latest price from the market
+        // (i.e. Chainlink provided price)
+        if (publishTime < block.timestamp - MAX_PRICE_LATENCY) {
+            priceOracle = PriceOracleUsed.CHAINLINK;
+            (price,) = _market.assetPrice();
```

Comment:
**Issue:** I think you need to handle the case here where `assetPrice()` returns an invalid price:
```solidity
        (uint256 price, bool invalid) = _market.assetPrice();
        if (invalid) {
            revert InvalidPrice();
        }
```

Reply:
Ah yes like before. Will add